server/src/main/java/io/crate/execution/jobs/PKLookupTask.java

```diff
@@ -98,6 +105,19 @@ public final class PKLookupTask extends AbstractTask {
 
         InputFactory.Context<CollectExpression<Doc, ?>> ctx = inputFactory.ctxForRefs(txnCtx, docRefResolver);
         ctx.add(toCollect);
+
+        sourceParser = new SourceParser();
+        for (Symbol symbol : toCollect) {
+            RefVisitor.visitRefs(
+                    symbol,
+                    ref -> {
+                        if (ref.column().isSystemColumn() == false && ref.granularity() == RowGranularity.DOC) {
+                            sourceParser.register(toSourceLookup(ref).column(), ref.valueType());
+                        }
+                    }
+            );
```

Comment:
There is an edge case of selecting _doc and regular column in a single query.

Could you try updating a test with such scenario to verify that _doc can see all columns?

Something like (https://github.com/crate/crate/pull/14665/commits/908bbcc666681a8a930c4b75ccb3a95c78e818dd#diff-dc89f0bea3e02a3afbff1c08f14546a984e60278688658124f864be8ad2bd484R1468-R1469)?

If it fails, I guess we need to add all table column into `toCollect`?

Reply:
Good catch! I've pushed a fixup to verify and fix it.